databricks/koalas/frame.py

```diff
@@ -7247,6 +7248,85 @@ def append(
 
         return cast(DataFrame, concat([self, other], ignore_index=ignore_index))
 
+    def combine_first(self, other: "DataFrame") -> "DataFrame":
+        """
+        Update null elements with value in the same location in `other`.
+
+        Combine two DataFrame objects by filling null values in one DataFrame
+        with non-null values from other DataFrame. The row and column indexes
+        of the resulting DataFrame will be the union of the two.
+
+        Parameters
+        ----------
+        other : DataFrame
+            Provided DataFrame to use to fill null values.
+
+        Returns
+        -------
+        DataFrame
+
+        See Also
+        --------
+        DataFrame.combine : Perform series-wise operation on two DataFrames
+            using a given function.
+
+        Examples
+        --------
+        >>> from databricks.koalas.config import set_option, reset_option
+        >>> set_option("compute.ops_on_diff_frames", True)
+        >>> df1 = ks.DataFrame({'A': [None, 0], 'B': [None, 4]})
+        >>> df2 = ks.DataFrame({'A': [1, 1], 'B': [3, 3]})
+        >>> df1.combine_first(df2)
+             A    B
+        0  1.0  3.0
+        1  0.0  4.0
+
+        Null values still persist if the location of that null value
+        does not exist in `other`
+
+        >>> df1 = ks.DataFrame({'A': [None, 0], 'B': [4, None]})
+        >>> df2 = ks.DataFrame({'B': [3, 3], 'C': [1, 1]}, index=[1, 2])
+        >>> df1.combine_first(df2)
+             A    B    C
+        0  NaN  4.0  NaN
+        1  0.0  3.0  1.0
+        2  NaN  3.0  1.0
+
+        >>> reset_option("compute.ops_on_diff_frames")
+        """
+
+        update_columns = set(self._internal.column_labels).intersection(
+            set(other._internal.column_labels)
+        )
+        final_spark_columns = []
+
+        combined_df = combine_frames(self, other)
+        column_labels = combined_df._internal.column_labels
+        updated_sdf = combined_df._internal.resolved_copy.spark_frame
+        spark_columns = combined_df._internal.spark_columns
+
+        for column_label in column_labels:
+            if (column_label[1],) in update_columns:
```

Comment:
Seems not valid for MultiIndex columns ??

Could you try with below `self` and `other` ??

```python
self = pd.DataFrame({'A': [None, 0], 'B': [None, 4]}, columns=pd.MultiIndex.from_tuples([('A', 'hello'), ('B', 'hi')]))
other = pd.DataFrame({'B': [3, 3], 'C': [1, 1]}, columns=pd.MultiIndex.from_tuples([('B', 'hi'), ('C', 'okay')]))
```

Reply:
right, would it make sense to use something like following
```
for column_label in column_labels:
            if column_label[1:] in update_columns:
                if column_label[0] == "this":
                    column_name = self._internal.spark_column_name_for(column_label[1:])
```